algorithms/unit_tests/TestStdAlgorithmsTeamAdjacentDifference.cpp

```diff
@@ -0,0 +1,206 @@
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include <TestStdAlgorithmsCommon.hpp>
+#include <cstddef>
+#include <functional>
+#include <numeric>
+#include "gtest/gtest.h"
+#include "std_algorithms/Kokkos_BeginEnd.hpp"
+
+namespace Test {
+namespace stdalgos {
+namespace TestAdjacentDifference {
+
+namespace KE = Kokkos::Experimental;
+
+template <class ValueType>
+struct PlusFunctor {
+  KOKKOS_INLINE_FUNCTION constexpr ValueType operator()(
+      const ValueType& lhs, const ValueType& rhs) const {
+    return lhs + rhs;
+  }
+};
+
+template <class ViewType, class DiffsViewType, class BinaryOp>
+struct TestFunctorA {
+  ViewType m_view;
+  DiffsViewType m_diffsView;
+  int m_apiPick;
+  BinaryOp m_binaryOp;
+
+  TestFunctorA(const ViewType view, const DiffsViewType diffsView, int apiPick,
+               BinaryOp binaryOp)
+      : m_view(view),
+        m_diffsView(diffsView),
+        m_apiPick(apiPick),
+        m_binaryOp(std::move(binaryOp)) {}
+
+  template <class MemberType>
+  KOKKOS_INLINE_FUNCTION void operator()(const MemberType& member) const {
+    const auto myRowIndex = member.league_rank();
+
+    auto myRowView  = Kokkos::subview(m_view, myRowIndex, Kokkos::ALL());
+    auto myDiffView = Kokkos::subview(m_diffsView, myRowIndex, Kokkos::ALL());
+
+    switch (m_apiPick) {
+      case 0: {
+        KE::adjacent_difference(member, KE::begin(myRowView),
```

Comment:
these should return an iterator, but you are ignoring it

Reply:
Done.